ibm/service/vpc/resource_ibm_is_subnet_public_gateway_attachment.go

```diff
@@ -0,0 +1,332 @@
+		Importer:      &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			isSubnetID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The subnet identifier",
+			},
+
+			isPublicGatewayID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of public gateway",
+			},
+
+			isPublicGatewayName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the Public gateway instance",
+			},
+
+			isPublicGatewayFloatingIP: {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+
+			isPublicGatewayStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway instance status",
+			},
+
+			isPublicGatewayResourceGroup: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway resource group info",
+			},
+
+			isPublicGatewayVPC: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway VPC info",
+			},
+
+			isPublicGatewayZone: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway zone info",
+			},
+
+			IsPublicGatewayResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			isPublicGatewayCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			flex.ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+		},
+	}
+}
+
+func resourceIBMISSubnetPublicGatewayAttachmentCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	subnet := d.Get(isSubnetID).(string)
+	publicGateway := d.Get(isPublicGatewayID).(string)
+
+	publicGatewayIdentity := &vpcv1.PublicGatewayIdentity{
+		ID: &publicGateway,
+	}
+
+	// Construct an instance of the SetSubnetPublicGatewayOptions
+	setSubnetPublicGatewayOptions := &vpcv1.SetSubnetPublicGatewayOptions{
+		ID:                    &subnet,
+		PublicGatewayIdentity: publicGatewayIdentity,
+	}
+
+	pg, response, err := sess.SetSubnetPublicGatewayWithContext(context, setSubnetPublicGatewayOptions)
+
+	if err != nil {
+		log.Printf("[DEBUG] Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response)
+		return diag.FromErr(fmt.Errorf("[ERROR] Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response))
+	}
+	d.SetId(subnet)
```

Comment:
should not ID be combination of subnetid/gatewayid

Reply:
Not required, subnet id is enough to retrieve the public gateway attachment. And only one public gateway can be attached.